byterun/unix.c

```diff
@@ -80,9 +81,18 @@ int caml_write_fd(int fd, int flags, void * buf, int n)
 {
   int retcode;
  again:
+#if defined(NATIVE_CODE) && defined(WITH_SPACETIME)
+  if ((flags & CHANNEL_FLAG_BLOCKING_WRITE)
+      == CHANNEL_FLAG_BLOCKING_WRITE) {
```

Comment:
You don't have to test for equality here because `CHANNEL_FLAG_BLOCKING_WRITE` is single bit. Just test for `flags & CHANNEL_FLAG_BLOCKING_WRITE` being non-zero.